### ISSUE TITLE
Expanding cell taking time

### ISSUE THREAD

**Issue by tapan-nathvani** [gleue/TGLStackedViewController#47]
in this method 
    - (void)collectionView:(UICollectionView *)collectionView didSelectItemAtIndexPath:
    (NSIndexPath *)indexPath {
    if (self.exposedItemIndexPath && indexPath.item == self.exposedItemIndexPath.item) {
        self.exposedItemIndexPath = nil;

}
else {
NSLog(@"Hello1");
        self.exposedItemIndexPath = indexPath;
        NSLog(@"Hello2");
}

So when i am assigning exposedItemIndexPath its taking almost 5 to 6 seconds to expand a cell.

Can you please help me to improve this ?

**Comment by gleue** [comment]
Having around 100 cards in the sample app makes selecting/deselecting noticeably slower. Recomputing the layout transitions takes its time.

How many cells do you have?

**Comment by tapan-nathvani** [comment]
I have almost 120+ cards and it will increase.

**Comment by gleue** [comment]
Hm, that could be a problem.

When switching layouts UIKit creates animations for all cards from one to the other layout. That would mean 78+ animations in parallel.

Could you try switching layouts w/o animations. Just to get an idea?

**Comment by tapan-nathvani** [comment]
Yes i tried by Passing NO to animation.. But it still taking time.

In this method of TGLStackedViewController Class

- (void)setExposedItemIndexPath:(nullable NSIndexPath *)exposedItemIndexPath {
    
    [self setExposedItemIndexPath:exposedItemIndexPath animated:NO];
}

is there any solution to make it fast ?

**Comment by gleue** [comment]
You could use instruments to see where the lag is introduced

**Comment by tapan-nathvani** [comment]
OKies. Will check it and let you know.
